enhancements/sdk-fbc-bundle-upgrade.md

```diff
@@ -0,0 +1,277 @@
+---
+title: sdk-fbc-run-bundle-upgrade
+authors:
+  - "@rashmigottipati"
+reviewers:
+  - "@joelanford"
+  - "@VenkatRamaraju"
+  - "@everettraven"
+  - "@jmrodri"
+approvers:
+  - "@joelanford"
+  - "@VenkatRamaraju"
+  - "@jmrodri"
+creation-date: 2022-04-26
+last-updated: 2022-04-26
+status: implementable
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# Add FBC support to run bundle-upgrade
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [X] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+- Should the FBC image be a required flag for bundle-upgrade?  
+
+  - The FBC or SQLite image can always be inferred from the annotations (for both `run bundle` and traditional OLM case).
+  - For the cases where the previous version of the operator was installed traditionally, we can infer the latest CSV from channel entries of the rendered index image.
+
+- We have bundle add mode that describes how a bundle is added to an index image; do we need those for FBC support? If so, are there any new modes for FBC other than `semver` and `replaces`?
+
+  - There are no add modes for FBC, hence, we should error out when a mode flag is specified and if the index image is a File-Based Catalog/Default Index image.
+
+## Summary
+
+The goal of this enhancement is to discuss how the `run bundle-upgrade` command in Operator SDK can begin to support and adopt the new format for indexes along with continuing to support the existing SQLite index images.
+
+## Motivation
+
+Currently, there is `run bundle-upgrade` command in the Operator SDK for upgrading an operator bundle to a newer version using OLM. OLM has transitioned to the new [File-Based Catalog (FBC)](https://olm.operatorframework.io/docs/reference/file-based-catalogs/#docs) for catalog images which is a more human readable and consumable format of a catalog that is in the form of `JSON/YAML`. As of now, Operator SDK's `run bundle-upgrade` command upgrades an operator bundle using [opm tooling]((<https://github.com/operator-framework/operator-registry/blob/master/docs/design/opm-tooling.md#opm>) that uses a SQLite database under the hood, which creates the registry database as well as index images. Support for these opm commands will be completely removed in a future release, hence Operator SDK should enable support for the new FBC format along with continuing to support the existing SQLite format so that Operator Authors can continue using SDK to perform and test operator upgrades on cluster using OLM for both formats.
```

Comment:
```suggestion
Currently, there is `run bundle-upgrade` command in the Operator SDK for upgrading an operator bundle to a newer version using OLM. OLM has transitioned to the new [File-Based Catalog (FBC)](https://olm.operatorframework.io/docs/reference/file-based-catalogs/#docs) for catalog images which is a more human readable and consumable format of a catalog that is in the form of `JSON/YAML`. As of now, Operator SDK's `run bundle-upgrade` command upgrades an operator bundle using [opm tooling](https://github.com/operator-framework/operator-registry/blob/master/docs/design/opm-tooling.md#opm) that uses a SQLite database under the hood, which creates the registry database as well as index images. Support for these opm commands will be completely removed in a future release, hence Operator SDK should enable support for the new FBC format along with continuing to support the existing SQLite format so that Operator Authors can continue using SDK to perform and test operator upgrades on cluster using OLM for both formats.
```

Reply:
Can't tell from the preview is my suggestion looks fine in markdown.

Reply:
Nope, it doesn't, this has to be wrapped around `<>` apparently.